server/middlewares/UserValidation.js

```diff
@@ -0,0 +1,77 @@
+import bcrypt from 'bcrypt';
+import CatchErrors from '../helper/CatchErrors';
+import dbConnect from '../connections/dbConnect';
+import { checkEmail } from '../helper/sqlHelper';
+
+const { catchTableReadError } = CatchErrors;
+/**
+  * @class UserValidation
+  *
+  * @description Validates input fields on the user input
+  */
+class UserValidation {
+  /**
+        * @static
+        *
+        * @param {object} req - The request payload sent to the middleware
+        * @param {object} res - The response payload sent back from the middleware
+        * @param {object} next - The call back function to resume the next middleware
+        *
+        * @returns {object} - status Message and the response object
+        *
+        * @description This method validates the user input
+        * @memberOf UserValidation
+        */
+  static validateUser(req, res, next) {
```

Comment:
To maintain the `Single Responsibility Principle` in your controllers, move validation for each route to a separate middleware. Let your controllers be focused on one task only.